File: HelpDesk.API/Models/Ticket.cs
namespace HelpDesk.API.Models
{
    public class Ticket
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public TicketStatus Status { get; set; } = TicketStatus.Open;

        public TicketPriority Priority { get; set; } = TicketPriority.Medium;

        public string CreatedBy { get; set; } = string.Empty;

        public string? AssignedTo { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<TicketComment> Comments { get; set; } = new List<TicketComment>();

    }
}

File: HelpDesk.Client/src/types/Ticket.ts
export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: "Open" | "InProgress" | "Resolved" | "Closed";
  priority: "Low" | "Medium" | "High" | "Critical";
  createdBy: string;
  assignedTo: string | null;
  createdAt: string;
}